### PR TITLE
[MAX] Add Flux1 implementation for Flux.1 pipeline

### PR DESCRIPTION
**Pull request by kkimmk**
### Overview
This PR introduces the core Flux1 model implementation required by the Flux1 pipeline. It adds the architecture package, model definition, configuration, and supporting layers (attention, embeddings, activations, normalizations) along with weight adapter utilities.

This PR is made by splitting Flux1 implementation from the original PR(https://github.com/modular/modular/pull/5786).

### What’s included
- New Flux1 architecture package under `max/python/max/pipelines/architectures/flux1/`.
- Supporting layers for attention, embeddings, activations, and normalization, used in Flux1 model.
- Weight adapter utilities for mapping HF checkpoints (weight_adapters.py).

### NOTES
This model implementation does NOT serve as a standalone executable model. It is expected to be executed within the Flux1 pipeline. (Will update the related PR link later)
Definition of the base abstract class MaxModel is defined in https://github.com/modular/modular/pull/5811

### CODE REVIEW

**Comment by pei0033** [comment]
I have read the CLA Document and I hereby sign the CLA

File: max/python/max/pipelines/architectures/flux1/layers/normalizations.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        bias: bool = True,
+        dtype: DType = DType.bfloat16,
+    ):
+        """Initialize adaptive layer normalization zero single module.
+
+        Args:
+            embedding_dim: Size of each embedding vector.
+            norm_type: Type of normalization to use ("layer_norm").
+            bias: Whether to use bias in linear projection.
+            device: Device to place the module on.
```
Comment (katelyncaldwell) [inline max/python/max/pipelines/architectures/flux1/layers/normalizations.py]: Remove doc string for absent device arg

File: max/python/max/pipelines/architectures/flux1/layers/flux_attention.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+                ),
+            ]
+        )
+
+    def forward(self, hidden_states: Tensor) -> Tensor:
+        """Apply feedforward network to hidden states.
+
+        Args:
+            hidden_states: Input hidden states.
+            *args: Additional positional arguments (unused).
```
Comment (katelyncaldwell) [inline max/python/max/pipelines/architectures/flux1/layers/flux_attention.py]: Is there a reason we have `**args` and `**kwargs` here?

File: max/python/max/pipelines/architectures/flux1/layers/flux_attention.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+        return hidden_states
+
+
+class FeedForward(Module[[Tensor], Tensor]):
+    def __init__(
+        self,
+        dim: int,
+        dim_out: int | None = None,
+        mult: int = 4,
+        activation_fn: str = "geglu",
```
Comment (katelyncaldwell) [inline max/python/max/pipelines/architectures/flux1/layers/flux_attention.py]: is this a typo? if not, we should update the docstring

File: max/python/max/pipelines/architectures/flux1/flux1.py [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+        self,
+        hidden_states: Tensor,
+        encoder_hidden_states: Tensor,
+        pooled_projections: Tensor,
+        timestep: Tensor,
+        img_ids: Tensor,
+        txt_ids: Tensor,
+        guidance: Tensor | None = None,
+        joint_attention_kwargs: dict[str, Any] | None = None,
+        controlnet_block_samples: Any | None = None,
```
Comment (katelyncaldwell) [inline max/python/max/pipelines/architectures/flux1/flux1.py]: I don't see this used anywhere — is it needed? Same with `controlnet_single_block_samples` and `controlnet_blocks_repeat` below.

File: max/python/max/pipelines/architectures/flux1/layers/embeddings.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+        hidden_states = self.linear_2(hidden_states)
+        return hidden_states
+
+
+class CombinedTimestepTextProjEmbeddings(Module[[Tensor, Tensor], Tensor]):
+    def __init__(
+        self,
+        embedding_dim: int,
+        pooled_projection_dim: int,
+        dtype: DType = DType.bfloat16,
```
Comment (katelyncaldwell) [inline max/python/max/pipelines/architectures/flux1/layers/embeddings.py]: do we need a `dtype` arg here?

**Review by katelyncaldwell — APPROVED** [review]
A few minor (non-blocking) comments, but looks good overall! We just need to work through this CI issue and then should be able to merge.

**Comment by pei0033** [comment]
The model codes were written based on the existing Diffusers code and the migration of the legacy module implementation to module_v3.
So, there were issues such as missing unused arguments and inadequate docstrings.
Beyond the review, I identified and resolved other overlooked issues.
Thank you.

**Comment by KCaverly** [comment]
!sync

File: max/python/max/pipelines/architectures/flux1/model.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+from max.pipelines.lib.interfaces.component_model import (  # type: ignore[import-not-found]
+    ComponentModel,
+)
+
+from .flux1 import FluxTransformer2DModel
+from .model_config import FluxConfig
+from .weight_adapters import convert_safetensor_state_dict
+
+
+class Flux1Model(ComponentModel):
```
Comment (tolgacangoz) [inline max/python/max/pipelines/architectures/flux1/model.py:30]: Isn't this Flux.1's backbone transformer, rather than itself? Could you elaborate on the naming here?

File: max/python/max/pipelines/architectures/flux1/model_config.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+from typing import Any, ClassVar
+
+from max.driver import Device
+from max.dtype import DType
+from max.graph import DeviceRef
+from max.pipelines.lib import MAXModelConfigBase, SupportedEncoding
+from pydantic import Field
+
+
+class FluxConfigBase(MAXModelConfigBase):
```
Comment (tolgacangoz) [inline max/python/max/pipelines/architectures/flux1/model_config.py:23]: https://github.com/modular/modular/pull/5813/changes#r2755926428

File: max/python/max/pipelines/architectures/flux1/model.py (hunk 6 above)
Comment (kkimmk) [reply max/python/max/pipelines/architectures/flux1/model.py:30]: Yes, it's the DiT model. Since this branch is merged, we are thinking of modifying it to `Flux1TransformerModel` in the later PR. What do you think?

Comment (tolgacangoz) [reply max/python/max/pipelines/architectures/flux1/model.py:30]: I agree :+1:.
